compute/cloud-client/src/main/java/compute/disks/CloneEncryptedDisk.java

```diff
@@ -0,0 +1,108 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CloneEncryptedDisk {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String project = "YOUR_PROJECT_ID";
+
+    // Name of the zone in which you want to create the disk.
+    String zone = "europe-central2-b";
+
+    // Name of the disk you want to create.
+    String diskName = "YOUR_DISK_NAME";
+
+    // The type of disk you want to create. This value uses the following format:
+    // "zones/{zone}/diskTypes/(pd-standard|pd-ssd|pd-balanced|pd-extreme)".
+    // For example: "zones/us-west3-b/diskTypes/pd-ssd"
+    String diskType = String.format("zones/%s/diskTypes/pd-ssd", zone);
+
+    // Size of the new disk in gigabytes.
+    int diskSizeGb = 10;
+
+    // A link to the disk you want to use as a source for the new disk.
+    // This value uses the following format:
+    // "projects/{project_name}/zones/{zone}/disks/{disk_name}"
+    String diskLink = String.format("projects/%s/zones/%s/disks/%s", "PROJECT_NAME", "ZONE",
+        "DISK_NAME");
+
+    // Customer-supplied encryption key used for encrypting data in the source disk.
+    // The data will be encrypted with the same key in the new disk.
+    byte[] encryptionKey = null;
+
+    createDiskFromCustomerEncryptedKey(project, zone, diskName, diskType, diskSizeGb, diskLink,
+        encryptionKey);
+  }
+
+  // Creates a zonal non-boot persistent disk in a project with the copy of data
+  // from an existing disk.
+  // The encryption key must be the same for the source disk and the new disk.
+  public static void createDiskFromCustomerEncryptedKey(String project, String zone,
+      String diskName, String diskType, int diskSizeGb, String diskLink, byte[] encryptionKey)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `disksClient.close()` method on the client to safely
+    // clean up any remaining background resources.
+    try (DisksClient disksClient = DisksClient.create()) {
+
+      // Create a disk and set the encryption key.
+      Disk disk = Disk.newBuilder()
+          .setZone(zone)
+          .setName(diskName)
+          .setType(diskType)
+          .setSizeGb(diskSizeGb)
+          .setSourceDisk(diskLink)
+          .setDiskEncryptionKey(CustomerEncryptionKey
+              .newBuilder()
+              .setRawKeyBytes(ByteString.copyFrom(encryptionKey))
+              .build())
+          .build();
+
+      // Wait for the insert disk operation to complete.
```

Comment:
```suggestion
      // Wait 3 minutes for the insert disk operation to complete.
```

Reply:
It doesn't necessarily have to be 3 mins. It's just a timeout right?

Reply:
Got it. Thanks for clarifying.